js/parts-more/RadialAxis.js

```diff
@@ -341,7 +341,39 @@ radialAxisMixin = {
         var axis = this, center = axis.center, chart = axis.chart, inverted = chart.inverted, value = options.value, reverse = options.reverse, end = axis.getPosition(value), background = axis.pane.options.background ?
             (axis.pane.options.background[0] ||
                 axis.pane.options.background) :
-            {}, innerRadius = background.innerRadius || '0%', outerRadius = background.outerRadius || '100%', x1 = center[0] + chart.plotLeft, y1 = center[1] + chart.plotTop, x2 = end.x, y2 = end.y, a, b, otherAxis, xy, tickPositions, ret;
+            {}, innerRadius = background.innerRadius || '0%', outerRadius = background.outerRadius || '100%', x1 = center[0] + chart.plotLeft, y1 = center[1] + chart.plotTop, x2 = end.x, y2 = end.y, a, b, otherAxis, xy, tickPositions, ret, shapeArgs;
+        // Crosshair logic
+        if (options.isCrosshair) {
+            if (axis.isCircular) {
+                if (!H.defined(value)) {
+                    // When the snap is set to false
+                    x2 = options.chartX;
+                    y2 = options.chartY;
+                    value = axis.translate(Math.atan2(y2 - y1, x2 - x1) -
+                        axis.startAngleRad, true);
+                }
+                else {
+                    // When the snap is set to true
+                    shapeArgs = options.point.shapeArgs;
+                    if (shapeArgs.start) {
+                        // Find a true value of the point based on the
+                        // angle
+                        value = axis.translate(options.point.rectPlotY, true);
+                    }
+                }
+                end = axis.getPosition(value);
+                x2 = end.x;
+                y2 = end.y;
+            }
```

Comment:
The same situation like in wrap section above (extract function).

Reply:
Made a similar changes to ones mentioned comment above. I have created getCrosshairPosition function and place crosshair logic there.